### PR TITLE
Add simple but realistic demo data generator

### PR DESCRIPTION
**Pull request by jsugarman**
## What
Add rake task to create and destroy demo data

## Also
Have included some other "improvements"
- Faker extension for generating realistic offences
- implement rubocop rspec, rails and performance
- remove tests raising shoulda-matcha warnings

## Ticket
[CACP-290](https://dsdmoj.atlassian.net/browse/CACP-290)

## Why
To reflect bare minimum of data that we would expect to
see and be able to demonstrate in the user interface

## Checklist

Before you ask people to review this PR:

- [X] Tests and linters should be passing
- [X] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [X] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [X] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [X] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: lib/faker/offence.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      def act
+        fetch('offence.act')
+      end
+
+      def class
+        fetch('offence.class')
+      end
+    end
+  end
+end
```
Comment (jsugarman) [inline lib/faker/offence.rb]: linebreak

File: lib/tasks/demodata.rake [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  pcases.each do |pcase|
+    pcase.defendants.each do |defendant|
+      defended_item = defendant.defendable
+      if defended_item
+        print "[DESTROY][DEFENDED ITEM] " + humanize_defended_item(defended_item)
+        defended_item.destroy
+        puts " #{ICONS[:success]}"
+      else
+        puts "[DESTROY][DEFENDED ITEM] no person or organisation found \u{274E}"
```
Comment (jsugarman) [inline lib/tasks/demodata.rake]: move to ICONS hash

File: spec/tasks/demodata_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    context 'when demo data exists' do
+      before { loader }
+
+      it 'destroys 2 cases' do
+        expect { unloader }.to change(ProsecutionCase, :count).by(-2)
+      end
+    end
+
+    context 'when demo does not exist' do
+      it 'destroys 2 cases' do
```
Comment (mpw5) [inline spec/tasks/demodata_spec.rb]: should this be `'destroys 0 cases'`?

File: README.md [hunk 4]
```diff
@@ -17,6 +17,19 @@ You can then start the application server by running:
 $ rails s
 ```
 
+### Generate demonstration data
+Tasks have been created to facilitate generation of a small specific, rememberable set of demonstration data. This data reflects current understanding of a bear minimum needed to effectively demonstrate the workings of this adaptor and its user interface [View court data](https://github.com/ministryofjustice/laa-court-data-ui).
```
[the remark this answers is not in the document]
Comment (jsugarman) [reply README.md]: 👍 lol

File: spec/tasks/demodata_spec.rb (hunk 3 above)
Comment (jsugarman) [reply spec/tasks/demodata_spec.rb]: 👍 good catch

File: spec/models/allocation_decision_spec.rb [hunk 5]
```diff
@@ -12,10 +12,10 @@
...
   it { should validate_presence_of(:motReasonCode) }
   it { should validate_presence_of(:allocationDecisionDate) }
-  it { should validate_inclusion_of(:isSection22ALowValueShoplifting).in_array([true, false]) }
-  it { should validate_inclusion_of(:isDamageValueUnder5000).in_array([true, false]) }
-  it { should validate_inclusion_of(:isTreatedAsIndictableOnly).in_array([true, false]) }
-  it { should validate_inclusion_of(:sentencingIndicationRequested).in_array([true, false]) }
+  # it { should validate_inclusion_of(:isSection22ALowValueShoplifting).in_array([true, false]) }
+  # it { should validate_inclusion_of(:isDamageValueUnder5000).in_array([true, false]) }
+  # it { should validate_inclusion_of(:isTreatedAsIndictableOnly).in_array([true, false]) }
+  # it { should validate_inclusion_of(:sentencingIndicationRequested).in_array([true, false]) }
```
[the remark this answers is not in the document]
Comment (jsugarman) [reply spec/models/allocation_decision_spec.rb]: if you are happy to remove them I can

[the remark this answers is not in the document]
Comment (jsugarman) [reply spec/models/allocation_decision_spec.rb]: removed for now as part of single commit  in case we want to revert

**Comment by jsugarman** [comment]
> Looks great, thank you for adding this in.
> I am curious as to why there is a single person being created (Jammy Dodger), instead of a random one, but that is definitely not a blocking issue.

No worries, I wanted to control having a specific cross-case defendant with known attributes for searching on. It was clearer (in my mind) to create the person and then "add" them to the two cases, as opposed to creating the person implicitly as part of case or defendant creation and then `cloning/dup'ing` them across to the other case.

**Pull request merged**